File: Java/dependency-injection/_03_control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Inbound/UdpSocket.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Inbound;

import java.util.Random;

public class UdpSocket implements InboundSocket {
  public boolean receive(byte[]frameData) {
    frameData = new byte[100];
    new Random().nextBytes(frameData);
    return true;
  }
}

File: CSharp/DomainSpecificLanguages/DomainSpecificLanguages/AlarmsNestedFunctions/Inbound/UdpSocket.cs
using System;

namespace TelecomSystemNestedFunctions.Inbound
{
  public interface IInboundSocket
  {
    bool Receive(out byte[] frameData);
  }

  public class UdpSocket : IInboundSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      new Random().NextBytes(frameData);
      return true;
    }
  }
}